discovery_engine_core/core/src/config.rs

```diff
@@ -164,11 +185,16 @@ pub(crate) fn de_config_from_json(json: &str) -> Figment {
         .join(Serialized::defaults(CoiSystemConfig::default()))
         .join(Serialized::default("core", CoreConfig::default()))
         .join(Serialized::default("endpoint", EndpointConfig::default()))
+        .join(Serialized::default(
+            &Exploration::id().to_string(),
+            ExplorationConfig::default(),
+        ))
```

Comment:
:grey_question:  not so familiar with `figment` and not related to your change, but just wondering why (unlike for the other configs) there is no key here for `CoiSystemConfig`?

Reply:
this is due to the way the `CoiSystemConfig` is layed out. it contains to fields `coi` and `kpe` which themselves hold configs. when reading the via `Figment`s, the whole json string is treated a the readable object and the two keys are mapped to the subconfigs of the `CoiSystemConfig`. this is also why it uses `Serialized::defaults(value)` instead of the `Serialized::default(key, value)` (mind the trailing "s").
we could also key the `CoiSystemConfig`, but then we would just need another nesting in the json config, which we avoid by the current handling.